api/v1alpha1/nifiresource_types.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	k8sv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Change the list to {"PROCESS_GROUP"} when all the type are available
+// +kubebuilder:validation:Enum={"PROCESS_GROUP"}
+type NiFiResourceType string
+
+const (
+	ProcessGroup NiFiResourceType = "PROCESS_GROUP"
+	// Processor          NiFiResourceType = "PROCESSOR"
+	// Connection         NiFiResourceType = "CONNECTION"
+	// InputPort          NiFiResourceType = "INPUT_PORT"
+	// OutputPort         NiFiResourceType = "OUTPUT_PORT"
+	// Funnel             NiFiResourceType = "FUNNEL"
+	// ControllerService  NiFiResourceType = "CONTROLLER_SERVICE"
+	// ReportingTask      NiFiResourceType = "REPORTING_TASK"
+	// RemoteProcessGroup NiFiResourceType = "REMOTE_PROCESS_GROUP"
+	// ParameterContext   NiFiResourceType = "PARAMETER_CONTEXT"
+)
```

Comment:
Should be moved in `common_types.go` like `ComponentType`, and should be renamed to `ResourceType`.

Reply:
And full uppercase is a bit aggressive and mismatch what it’s done for the other